optaplanner-examples/src/main/java/org/optaplanner/examples/batchscheduling/domain/Segment.java

```diff
@@ -0,0 +1,77 @@
+package org.optaplanner.examples.batchscheduling.domain;
+
+import org.optaplanner.examples.common.domain.AbstractPersistable;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+
+@XStreamAlias("PipeSegment")
+public class Segment extends AbstractPersistable {
+
+    private Batch batch;
+    private RoutePath routePath;
+    private String name;
+    private Integer sequence;
```

Comment:
Any reason why this is not `int`? `Integer` takes up much more memory than the primitive `int`.
(That said, you may need this nullable, in which case you don't really have another option.)